luigi_pipeline/lib/model/gcnv_mt_schema.py

```diff
@@ -0,0 +1,212 @@
+        lof = hl.len(hl_agg_collect_set_union(parse_genes(self.mt.genes_LOF_Ensemble_ID))) > 0
+        copy_gain = hl.len(hl_agg_collect_set_union(parse_genes(self.mt.genes_CG_Ensemble_ID))) > 0
+        return hl.case().when(
+            lof & copy_gain, 
+            [formatted_gcnv_consequence, "LOF", "COPY_GAIN"]
+        ).when(
+            lof, 
+            [formatted_gcnv_consequence, "LOF"]
+        ).when(
+            copy_gain, 
+            [formatted_gcnv_consequence, "COPY_GAIN"]
+        ).default(
+            [formatted_gcnv_consequence]
+        )
+
+    @row_annotation(name='sortedTranscriptConsequences', fn_require=gene_ids)
+    def sorted_transcript_consequences(self):
+        lof_genes = hl_agg_collect_set_union(parse_genes(self.mt.genes_LOF_Ensemble_ID))
+        copy_gain_genes = hl_agg_collect_set_union(parse_genes(self.mt.genes_CG_Ensemble_ID))
+        major_consequence_genes = lof_genes | copy_gain_genes
+        return hl.map(
+            lambda gene: hl.if_else(
+                major_consequence_genes.contains(gene),
+                {
+                    "gene_id": gene, 
+                    "major_consequence": hl.if_else(
+                        lof_genes.contains(gene),
+                        "LOF",
+                        "COPY_GAIN",
+                    )
+                },
+                {"gene_id": gene},
+            ),
+            self.mt.geneIds,
+        )
+
+    @row_annotation(fn_require=start)
+    def pos(self):
+        return self.mt.start
+
+    @row_annotation(fn_require=[contig, pos])
+    def xpos(self):
+        return variant_id.get_expr_for_xpos(
+            hl.locus(self.mt.contig, self.mt.pos)
+        )
+
+    @row_annotation(fn_require=xpos)
+    def xstart(self):
+        return self.mt.xpos
+
+    @row_annotation(fn_require=[contig, end])
+    def xstop(self):
+        return variant_id.get_expr_for_xpos(
+            hl.locus(self.mt.contig, self.mt.end)
+        )
+
+class SeqrGCNVGenotypesSchema(SeqrGenotypesSchema):
+
+    def __init__(self, *args, is_new_joint_call=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._is_new_joint_call = is_new_joint_call
+
+    @row_annotation(fn_require=SeqrGenotypesSchema.genotypes)
+    def samples(self):
+        return self._genotype_filter_samples(lambda g: True)
+
+    @row_annotation(fn_require=SeqrGenotypesSchema.genotypes)
+    def samples_new_call(self):
+        return self._genotype_filter_samples(lambda g: g.new_call)
+
+    def samples_no_call(self):
+        pass
+
+    def samples_num_alt(self):
+        pass
+
+    def samples_gq(self):
+        pass
+
+    def samples_ab(self):
+        pass
+
+    @row_annotation(fn_require=SeqrGenotypesSchema.genotypes)
+    def samples_qs(self, start=0, end=1000, step=10):
+        return hl.struct(**{
+            '%i_to_%i' % (i, i+step): self._genotype_filter_samples(lambda g: ((g.qs >= i) & (g.qs < i+step)))
```

Comment:
I think we try to use f-string for string formatting, but I defer to Mike on style for this repo

Reply:
Yep I agree!  All the existing bucketer functions look like this so I kept as is for now.